src/backend/distributed/operations/citus_tools.c

```diff
@@ -54,12 +57,40 @@ static void ExecuteCommandsAndStoreResults(StringInfo *nodeNameArray,
 										   int commandCount);
 static bool ExecuteRemoteQueryOrCommand(char *nodeName, uint32 nodePort,
 										char *queryString, StringInfo queryResult);
+static bool ExecuteRemoteQueryOrCommandUsingExistingConnections(char *nodeName,
+																uint32 nodePort,
+																char *queryString,
+																StringInfo
+																queryResultString);
+static bool ExecuteRemoteQueryOrCommandInternal(char *nodeName, uint32 nodePort,
+												char *queryString,
+												StringInfo queryResultString,
+												int connectionFlags);
 static Tuplestorestate * CreateTupleStore(TupleDesc tupleDescriptor,
 										  StringInfo *nodeNameArray, int *nodePortArray,
 										  bool *statusArray,
 										  StringInfo *resultArray, int commandCount);
 
 
+/*
+ * worker_check_connection_to_node sends a simple query from a worker node to another
+ * node, and returns success status.
+ */
+Datum
+worker_check_connection_to_node(PG_FUNCTION_ARGS)
```

Comment:
Does the name should include `worker`? I think given that we might be connecting to coordinator (or in the future re-name all these concepts), `check_connection_to_node` might be better. 

Also, should it be `health_check_to_node` or such?  I guess given the current implementation, we'd only check connectivity, but health check might mean a lot, see [here](https://bucardo.org/check_postgres/check_postgres.pl.html#cluster_id).

So, if we think that we might expand this ini the future to check more stuff, `health_check_to_node` might be better suited.

Reply:
Also, we might consider moving the logic into a C function (e.g., `HealthCheckForNode()`) or such, because the higher level UDF (e.g., go over all the nodes and call this) might rely on that C function. 

This UDF is not going to be called at the highest level UDF that we run over the coordinator.

Reply:
I moved the logic into a C function. However I think we may need to send the UDF call from coordinator to all the worker nodes.

Even if we do not use the C function directly, it still adds value as it separates the parsing of SQL arguments and the actual logic for checking connections.